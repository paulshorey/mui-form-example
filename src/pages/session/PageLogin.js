/* core */
import React from 'react';
/* components */
import Box from 'components/box/Box';

/* forms */
import { MuiForm, MuiInput } from 'mui-form';
import * as Styled from './PageLoginStyled';

/*
	form
*/
class PageComponent extends React.Component {
	constructor() {
		super();
		this.state = {
			muiFormButtons: {
				submit: ' Enter',
				reset: false,
			},
		};
	}

	render() {
		var { history } = this.props;

		setTimeout(()=>{
			window.store.message = {
				title: <span><span className="icon-warning"></span> This app is not-functional without the Mui hardware. <a href="/XWC1001/examples/validation" style={{fontWeight:"bold"}} className="color_success fix_textWrap">CLICK HERE TO ENTER the DEMO &raquo;</a> </span>,
				type: 'warning'
			};
		},3000);

		var formSubmit = valid => {
			console.warn('submitting...');
			this.setState({ muiFormSubmitting: true });
			window.store.message = {};
			valid
				.then(event => {
					// form is valid
					this.props.ubus
						.login(this.state.muiFormValues)
						.then(action => {
							this.setState({ muiFormSubmitting: false });
							history.push('/' + window.store.deviceInfo.model + '/status/controller');
						})
						.catch(err => {
							// dispatch(layoutActions.message({title:"Login failed. Please check that your device is connected and powered on.", status:1}));
							this.setState({ muiFormSubmitting: false });
							window.store.message = {
								title: 'Login failed. Please check that your device is connected and powered on.',
								type: 'error',
							};
						});
				})
				.catch(err => {
					// form is NOT valid
					console.error('Error: ' + err);
				});
		};

		var HelpBox = (
			<Box box={{ title: 'Login Help' }}>
				<div className="form" style={{ minWidth: '200px' }}>
					<p>
						Username: root<br />Password: admin
					</p>
				</div>
			</Box>
		);
		const TopChildren = [
			<span
				key="Login Help"
				className="fontIcon icon-pk_question"
				onClick={() => {
					window.store.side = { Box: HelpBox };
				}}
			/>,
		];

		return (
			<Styled.PageLogin>
				<Box box={{ title: 'Login', id: 'boxLogin', TopChildren: TopChildren }}>
					<MuiForm stateScope={this} onSubmit={formSubmit}>
						<div className="formSection">
							<fieldset>
								<label>Username:</label>
								<MuiInput stateScope={this} name="username" type="text" />
							</fieldset>
							<fieldset>
								<label>Password:</label>
								<MuiInput stateScope={this} name="password" type="password" />
							</fieldset>
						</div>
					</MuiForm>
				</Box>
			</Styled.PageLogin>
		);
	}
}

export default PageComponent;
