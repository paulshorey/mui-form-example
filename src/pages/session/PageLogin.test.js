import React from 'react';
import renderer from 'react-test-renderer';

// shallow + redux
import { shallow, connected, shallowConnected } from 'index.testing';
import PageLogin from './PageLogin';

it('renders correctly', () => {
	const tree = renderer.create(shallowConnected(<PageLogin />)).toJSON();
	expect(tree).toMatchSnapshot();
});
