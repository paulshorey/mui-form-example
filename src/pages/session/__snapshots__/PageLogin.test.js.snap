// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-iwsKbI gnbXsY"
>
  <div
    className="sc-gzVnrw bYkUlK"
  >
    <div
      className="messages_container"
    >
      <div
        className="messages_content"
      >
        <div
          className="message"
          onMouseEnter={[Function]}
        >
          <div
            className="message_content"
          >
            <div
              className="messageX"
              onClick={[Function]}
            >
              <span
                className="fontIcon icon-cross"
                title="hide message"
              />
            </div>
            <div
              className="message_text messageTitle"
            >
              <div>
                <span>
                   
                </span>
                 
                
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="layout_container noNav"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="sc-bwzfXH Xhbhr"
    >
      <div>
        <div
          className="sc-ifAKCX bIPvnp"
          id="Hamburger"
          onClick={[Function]}
        >
          <div
            className="hamburger-icon"
          >
            <div
              className="hamburger-centered"
            >
              <div
                className="hamburger-top"
              />
              <div
                className="hamburger-middle"
              />
              <div
                className="hamburger-bottom"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-htpNat ipbLAt"
        >
          <a
            href="/"
          >
            <img
              alt="LUXUL"
              src="/branding/logo/logo_white.svg"
            />
          </a>
        </div>
        <div
          className="sc-bxivhb kLTdtH"
          onClick={[Function]}
        >
          <span
            className="link"
          >
            <span
              className="title"
              style={
                Object {
                  "paddingTop": "0.1rem",
                  "textIndent": "-1.5px",
                }
              }
            >
              
            </span>
            <span
              className="text"
              style={
                Object {
                  "textIndent": "-1px",
                }
              }
            >
              Firmware: 
              
            </span>
          </span>
          <span
            className="fontIcon icon-navlink_info"
          />
        </div>
      </div>
    </div>
    <div
      className="layout_content"
    >
      <div
        className="page"
        onClick={[Function]}
      >
        <div
          className="sc-jTzLTM dwhbRT"
          onClick={[Function]}
        >
          <div
            className="box  sc-bdVaJa eujpTF"
            id="boxLogin"
          >
            <div
              className="box_content"
            >
              <div
                className="box_top"
              >
                <div
                  className="items"
                >
                  <div
                    className="item title"
                  >
                    Login
                  </div>
                  <span
                    className="fontIcon icon-pk_question"
                    onClick={[Function]}
                  />
                </div>
              </div>
              <div
                className="box_middle"
              >
                <div
                  className="MuiForm_customScrollbars sc-gqjmRU fALeYx"
                >
                  <form
                    autoComplete="off"
                    className="MuiForm sc-VigVT eoLjKH"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onSubmit={[Function]}
                    style={
                      Object {
                        "overflowY": "scroll",
                      }
                    }
                  >
                    <div
                      className="formSection"
                    >
                      <fieldset>
                        <label>
                          Username:
                        </label>
                        <div
                          className="control MuiInput"
                        >
                          <input
                            name="username"
                            onBlur={[Function]}
                            onChange={[Function]}
                            type="text"
                            value=""
                          />
                          
                        </div>
                      </fieldset>
                      <fieldset>
                        <label>
                          Password:
                        </label>
                        <div
                          className="control MuiInput"
                        >
                          <input
                            name="password"
                            onBlur={[Function]}
                            onChange={[Function]}
                            type="password"
                            value=""
                          />
                          
                        </div>
                      </fieldset>
                    </div>
                    <div
                      className="formSubmitSpacer"
                    />
                    <div
                      className="formSubmit"
                    >
                      <button
                        className="MuiButton  primary sc-gZMcBi clWKUC"
                        disabled={true}
                        type="submit"
                      >
                        <b>
                           Enter
                        </b>
                      </button>
                    </div>
                    <button
                      className="MuiButton clear scrollCue sc-gZMcBi clWKUC"
                      onClick={[Function]}
                      type="button"
                    >
                      <b>
                        <span
                          className="fontIcon icon-carat_down"
                        />
                      </b>
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="Bottom sc-EHOje fNRtIu"
    >
      <span
        className="logo"
      >
        <img
          alt="Legrand"
          src="/branding/logo/legrand_white.svg"
        />
      </span>
      <span
        className="copyright"
      >
        <span
          className="desktop_only"
        >
          This app is non-functional without the Mui hardware. 
        </span>
        <a
          className="color_success"
          href="/XWC1001/examples/validation"
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          CLICK HERE TO DEMO the UI »
        </a>
      </span>
    </div>
    <div
      className="sideBox  sc-htoDjs dXTkss"
    >
      <div
        className="sideBox_overlay"
        onClick={[Function]}
      />
      <div
        className="sideBox_content"
      >
        <div
          className="sideBox_close"
          onClick={[Function]}
        >
          <span
            className="fontIcon icon-cross"
            title="Close"
          />
        </div>
      </div>
    </div>
    <div
      className="popupBox  sc-dnqmqq feMKQw"
    >
      <div
        className="popupBox_overlay"
        onClick={[Function]}
      />
      <div
        className="popupBox_content"
      >
        <div
          className="popupBox_close"
          onClick={[Function]}
        >
          <span
            className="fontIcon icon-cross"
            title="Close"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
